src/games.c

```diff
@@ -1691,7 +1691,28 @@ int igraph_callaway_traits_game (igraph_t *graph, igraph_integer_t nodes,
     igraph_real_t maxcum;
     igraph_vector_t nodetypes;
 
-    /* TODO: parameter checks */
+    //parameter check 
+    if(nodes < 0){
+        IGRAPH_ERROR("Invalid number of vertices", IGRAPH_EINVAL);
+    }
+    
+    if (types < 1) {
+        IGRAPH_ERROR("types must be >= 1", IGRAPH_EINVAL);
```

Comment:
Again, it's best to spell out the problem: "The number of vertex types must be at least 1". This message will be seen by users of high-level interfaces, and it may not be clear at all to them what "types" means, as there may not be a parameter with this name.